### PR TITLE
Add support of Thanos Store grpc-service protocol to make able connect Promscale to Thanos Query

### PR DESCRIPTION
**Pull request by 2nick**
> 1. How is this helpful?

This PR adds functionality that allows using Promscale as part of [Thanos](https://github.com/thanos-io/thanos/) infrastructure.

Thanos is usually used with massive Prometheus installations and allows to run queries with metrics located on different Prometheus instances.

Connecting Promscale to Thanos Query component allows making queries that use metrics stored in TimescaleDB.

> 2. Aims behind the implementation. Are you trying to achieve something?

At first, I'd tried to use remote-read in Promscale with Thanos Sidecar which allows adding Prometheus instance to Thanos infrastructure.

But there is a problem that Thanos Sidecar trying to read Prometheus config through [/api/v1/status/config](https://prometheus.io/docs/prometheus/latest/querying/api/#config) to get external labels and Promscale has no such endpoint. Also sidecar won't start without external labels.

The main problem is that when Prometheus has external labels it will add them to samples when sends the stream to remote write and Thanos Sidecar add them when sends request to remote read. 

And if you had added some metrics without those external labels directly to TimescaleDB you wouldn't select them because Thanos Sidecar will add external labels to chunks select request.

For me it looks like adding the functionality of Thanos Store to Promscale is the best option here. :)

### CODE REVIEW

**Comment by 2nick** [comment]
@Harkishen-Singh could you please take look at this description and PR? :)

**Comment by cevian** [comment]
@2nick Thaks for this PR. I am going to get in touch with some Thanos developers to see if we can get some thoughts on this PR as well. I'll update shortly.

Thank you, Mat

File: pkg/runner/runner.go [hunk 1]
```diff
@@ -60,6 +64,25 @@ func Run(cfg *Config) error {
 	log.Info("msg", "Starting up...")
 	log.Info("msg", "Listening", "addr", cfg.ListenAddr)
 
+	srv := thanos.NewStorage(client.Queryable())
+	grpcServer := grpc.NewServer()
+	storepb.RegisterStoreServer(grpcServer, srv)
+
+	go func() {
+		log.Info("msg", fmt.Sprintf("Start listening %s", cfg.GrpcListenAddr))
```
Comment (cevian) [inline pkg/runner/runner.go:72]:
```suggestion
		log.Info("msg", fmt.Sprintf("Start listening for Thanos StoreAPI on %s", cfg.GrpcListenAddr))
```

File: pkg/runner/runner.go [hunk 2]
```diff
@@ -60,6 +64,25 @@ func Run(cfg *Config) error {
...
+		listener, err := net.Listen("tcp", cfg.GrpcListenAddr)
+		if err != nil {
+			log.Error("msg", "Listening grpc-server", "err", err)
```
Comment (cevian) [inline pkg/runner/runner.go:75]:
```suggestion
			log.Error("msg", "Listening for Thanos StoreAPI failed", "err", err)
```

File: pkg/runner/runner.go [hunk 3]
```diff
@@ -60,6 +64,25 @@ func Run(cfg *Config) error {
...
+			return
+		}
+
+		log.Info("msg", "Start thanos-store")
+		if err := grpcServer.Serve(listener); err != nil {
+			log.Error("msg", "Starting grpc-server", "err", err)
```
Comment (cevian) [inline pkg/runner/runner.go:81]:
```suggestion
			log.Error("msg", "Starting the Thanos store failed", "err", err)
```

File: go.mod [hunk 4]
```diff
@@ -33,9 +31,13 @@ require (
 	github.com/prometheus/prometheus v1.8.2-0.20210605142932-7bc11dcb0664
 	github.com/schollz/progressbar/v3 v3.7.2
 	github.com/sergi/go-diff v1.0.0
-	github.com/shurcooL/vfsgen v0.0.0-20200824052919-0d455de96546 // indirect
```
Comment (cevian) [inline go.mod:36]: why are these indirect dependencies removed

**Review by cevian — CHANGES_REQUESTED** [review]
This looks great. Only small changes requested. Will approve and merge once the changes are implemented. Thanks @2nick.

File: go.mod (hunk 4 above)
Comment (2nick) [reply go.mod:36]: Don't know, probably by go mod tidy

File: pkg/runner/flags.go [hunk 5]
```diff
@@ -56,6 +57,7 @@ func ParseFlags(cfg *Config, args []string) (*Config, error) {
 
 	fs.StringVar(&cfg.ConfigFile, "config", "config.yml", "YAML configuration file path for Promscale.")
 	fs.StringVar(&cfg.ListenAddr, "web-listen-address", ":9201", "Address to listen on for web endpoints.")
+	fs.StringVar(&cfg.GrpcListenAddr, "grpc-listen-address", ":9202", "Address to listen on for Thanos Store endpoints.")
```
[the remark this answers is not in the document]
Comment (2nick) [reply pkg/runner/flags.go:60]: Should I change config field name too?

[the remark this answers is not in the document]
Comment (cevian) [reply pkg/runner/flags.go:60]: that would be great if you could

**Comment by 2nick** [comment]
Actualized to 0.5.0

**Review by cevian — APPROVED** [review]

**Comment by cevian** [comment]
@andersje there is a conflict in go.sum if you resolve that then do a "bors r+" in comments to merge. **Thanks again**

**Comment by ramonguiu** [comment]
@2nick thanks so much for helping improve Promscale to be integrated in a Thanos installation.
Would you be interested in contributing a post on the Timescale blog (with our help) on the problem you wanted to solve, the process to get everything set up and the value you get from it?

**Pull request merged**